crates/blockchain/blockchain.rs

```diff
@@ -28,10 +28,18 @@ use ethereum_rust_vm::{
 ///
 /// Performs pre and post execution validation, and updates the database with the post state.
 pub fn add_block(block: &Block, storage: &Store) -> Result<(), ChainError> {
-    // TODO(#438): handle cases where blocks are missing between the canonical chain and the block.
+    let block_hash = block.header.compute_block_hash();
+
+    if storage.get_invalid_block(block_hash)?.is_some() {
```

Comment:
hmm, does it make sense to store invalid blocks?

Reply:
If that's the case, we could use some documentation (or link) that explains it

Reply:
That makes sense, we can add it again only after a feature requires it. Removed all references to invalid block tracking for now.